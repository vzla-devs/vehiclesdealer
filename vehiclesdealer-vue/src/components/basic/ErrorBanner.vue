<template>
  <div>
    <v-alert
      class="error-banner red"
      :value="true"
      dismissible
      type="error"
      @input="$emit('onClose')"
    >
      {{ message }}
    </v-alert>
  </div>
</template>

<script>
export default {
  props: {
    message: { type: String, required: true }
  }
}
</script>
