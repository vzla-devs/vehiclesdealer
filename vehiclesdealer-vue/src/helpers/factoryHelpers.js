import { shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'

export function wrapperBuilderFactory ({ component, localVue, stubs }) {
  let propsData = {}
  let store

  function withProps (props) {
    propsData = { ...props }
    return self
  }

  function withStore (newStore) {
    store = newStore
    return self
  }

  function build () {
    store = store ? new Vuex.Store({ ...store }) : undefined
    return shallowMount(component, { localVue, stubs, propsData, store })
  }

  const self = {
    withProps,
    withStore,
    build
  }
  return self
}