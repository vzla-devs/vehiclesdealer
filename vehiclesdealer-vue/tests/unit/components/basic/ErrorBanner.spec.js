import ErrorBanner from '@/components/basic/ErrorBanner.vue'
import { componentBuilder } from '@tests/helpers/builderHelpers'

describe('ErrorBanner.vue', () => {
  it('shows an error message', () => {
    const message = 'anyMessage'

    const wrapper = anErrorBanner().withProps({ message }).build()

    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits an event when closing the banner', () => {
    const wrapper = anErrorBanner().build()

    wrapper.find('.error-banner').vm.$emit('input')

    expect(wrapper.emitted().onClose).toBeTruthy()
  })

  function anErrorBanner () {
    const stubs = { 'v-alert': true }
    return componentBuilder(ErrorBanner).withStubs(stubs).withProps({ message: '' })
  }
})
