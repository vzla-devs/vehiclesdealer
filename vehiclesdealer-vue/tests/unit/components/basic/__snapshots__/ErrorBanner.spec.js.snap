// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBanner.vue shows an error message 1`] = `
<div>
  <v-alert-stub
    class="error-banner red"
    dismissible=""
    type="error"
    value="true"
  >
    
    anyMessage
  
  </v-alert-stub>
</div>
`;
